commit 342af93b654253a32f924d3d88dc3e1b4f449736 (refs/changes/25/1825/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-04-21T16:37:52-07:00 (2 years, 6 months ago)
    
    SAPI-286 SAPI DeleteInstance should cope with VMAPI DeleteVm returning 200 and no job_uuid

diff --git a/lib/server/vmapiplus.js b/lib/server/vmapiplus.js
index ca0d18e..41832ef 100644
--- a/lib/server/vmapiplus.js
+++ b/lib/server/vmapiplus.js
@@ -82,17 +82,29 @@ function deleteVm(uuid, cb) {
     var params = {};
     params.uuid = uuid;
 
-    vmapi.deleteVm(params, function (err, res) {
+    vmapi.deleteVm(params, function (err, body) {
         if (err) {
             log.error(err, 'failed to delete VM %s', uuid);
             return (cb(err));
+        } else if (!body.job_uuid) {
+            /*
+             * For a period of time (between ZAPI-685 and ZAPI-782), VMAPI
+             * DeleteVm could return a "200 OK" with a response body that
+             * did not include a "job_uuid" -- because no destroy job was
+             * created.
+             */
+            log.info({body: body}, 'VMAPI DeleteVm response: no job_uuid, ' +
+                'VM was synchronously marked deleted');
+            cb();
+            return;
         }
 
-        log.info({ job: res.job_uuid }, 'destroy job dispatched');
+        log.info({ vm_uuid: body.vm_uuid, job: body.job_uuid },
+            'destroy job dispatched');
 
         var expected_errors = [ 'ResourceNotFoundError' ];
 
-        waitForJob.call(self, res.job_uuid, expected_errors,
+        waitForJob.call(self, body.job_uuid, expected_errors,
             function (suberr) {
             if (suberr) {
                 suberr = new mod_errors.DestroyFailedError(
